sql/core/src/main/scala/org/apache/spark/sql/GroupedData.scala

```diff
@@ -34,15 +65,32 @@ import org.apache.spark.sql.types.NumericType
  * @since 1.3.0
  */
 @Experimental
-class GroupedData protected[sql](df: DataFrame, groupingExprs: Seq[Expression]) {
-
+class GroupedData protected[sql](
+    df: DataFrame,
+    groupingExprs: Seq[Expression],
+    private val groupType: GroupedData.GroupType) {
   private[sql] implicit def toDF(aggExprs: Seq[NamedExpression]): DataFrame = {
```

Comment:
actually since this is only used in one function, can we just remove this and inline it? i don't want to have too many implicits, especially when it doesn't provide a lot of benefits.